### ISSUE TITLE
Test failure System.IO.Tests.DirectoryInfo_SymbolicLinks.EnumerateFiles_LinksWithCycles_ThrowsTooManyLevelsOfSymbolicLinks(recurse: False)

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#54530]
Run: [runtime-coreclr libraries-jitstress 20210621.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1196911&view=ms.vss-test-web.build-test-results-tab&runId=35888492&resultId=144039&paneView=debug)

Failed test:
```
net6.0-Linux-Release-arm64-CoreCLR_checked-jitstress1_tiered-(Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- System.IO.Tests.DirectoryInfo_SymbolicLinks.EnumerateFiles_LinksWithCycles_ThrowsTooManyLevelsOfSymbolicLinks(recurse: False)

```
**Error message:**
```
System.IO.IOException : Invalid pipe handle.


Stack trace
   at System.IO.Pipes.PipeStream.ValidateHandleIsPipe(SafePipeHandle safePipeHandle) in /_/src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs:line 240
   at System.IO.Pipes.AnonymousPipeClientStream.Init(PipeDirection direction, SafePipeHandle safePipeHandle) in /_/src/libraries/System.IO.Pipes/src/System/IO/Pipes/AnonymousPipeClientStream.cs:line 73
   at System.IO.Pipes.AnonymousPipeClientStream..ctor(PipeDirection direction, SafePipeHandle safePipeHandle) in /_/src/libraries/System.IO.Pipes/src/System/IO/Pipes/AnonymousPipeClientStream.cs:line 66
   at System.Diagnostics.Process.OpenStream(Int32 fd, PipeDirection direction) in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 787
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 481
   at System.Diagnostics.Process.Start() in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs:line 1240
   at MountHelper.CreateSymbolicLink(String linkPath, String targetPath, Boolean isDirectory) in /_/src/libraries/System.IO.FileSystem/tests/PortedCommon/ReparsePointUtilities.cs:line 49
   at System.IO.Tests.BaseSymbolicLinks.CreateDirectoryContainingSelfReferencingSymbolicLink() in /_/src/libraries/System.IO.FileSystem/tests/Base/BaseSymbolicLinks.cs:line 14
   at System.IO.Tests.DirectoryInfo_SymbolicLinks.EnumerateFiles_LinksWithCycles_ThrowsTooManyLevelsOfSymbolicLinks(Boolean recurse) in /_/src/libraries/System.IO.FileSystem/tests/DirectoryInfo/SymbolicLinks.cs:line 40
```

**Comment by danmoseley** [comment]
@carlossanlop

**Comment by adamsitnik** [comment]
Fixed by @danmoseley in #54660
